go/def.bzl

```diff
@@ -389,9 +390,24 @@ def emit_go_link_action(ctx, importmap, transitive_libs, cgo_deps, lib,
   # Avoided -s on OSX but but it requires dsymutil to be on $PATH.
   # TODO(yugui) Remove this workaround once rules_go stops supporting XCode 7.2
   # or earlier.
-  cmds += ["export PATH=$PATH:/usr/bin"]
   cmds += [
+    "export PATH=$PATH:/usr/bin",
     "export GOROOT=$(pwd)/" + ctx.file.go_tool.dirname + "/..",
+    "STAMP_XDEFS=()",
+  ]
+
+  stamp_input = []
+  if ctx.attr.stamp and ctx.attr.linkstamp:
+    stamp_input = [ctx.info_file, ctx.version_file]
+    for f in stamp_input:
+      cmds += [
+          "while read -r stamp || [[ -n $stamp ]]; do",
+          "if echo $stamp | grep -qe '^.*[[:space:]].*$'; then",
+          "STAMP_XDEFS+=(-X $(echo %s.${stamp} | tr ' ' '='));" % ctx.attr.linkstamp,
```

Comment:
if `linkstamp` is empty, this results in `-X .BUILD_TIMESTAMP=1234` - is what what we want?
also, is it legal for values to have spaces? do we need to add quotes around the values?